lightning/src/chain/keysinterface.rs

```diff
@@ -178,9 +184,9 @@ impl ChannelKeys for InMemoryChannelKeys {
 	fn htlc_base_key(&self) -> &SecretKey { &self.htlc_base_key }
 	fn commitment_seed(&self) -> &[u8; 32] { &self.commitment_seed }
 
-	fn sign_remote_commitment<T: secp256k1::Signing>(&self, channel_value_satoshis: u64, channel_funding_script: &Script, feerate_per_kw: u64, commitment_tx: &Transaction, keys: &TxCreationKeys, htlcs: &[&HTLCOutputInCommitment], to_self_delay: u16, secp_ctx: &Secp256k1<T>) -> Result<(Signature, Vec<Signature>), ()> {
+	fn sign_remote_commitment<T: secp256k1::Signing>(&self, channel_value_satoshis: u64, channel_funding_redeemscript: &Script, feerate_per_kw: u64, commitment_tx: &Transaction, keys: &TxCreationKeys, htlcs: &[&HTLCOutputInCommitment], to_self_delay: u16, secp_ctx: &Secp256k1<T>) -> Result<(Signature, Vec<Signature>), ()> {
```

Comment:
Implemented with parameter as `feerate_per_kw` instead of `asbolute_fee`. I think we also conclude verbally than fee passed by caller can't be trusted by signer and should be recomputed locally.

Reply:
Hmm, no? Indeed, the parameter name in the trait is wrong, its per_kw, not absolute, but we can't not provide it - its required to generate the commitment transaction itself. Indeed, we can't trust it, and either our peer or our own fee estimator can make us attach an absurd fee and spend all our money on fee, but until the protocol doesn't have an update_fee message, we can't do anything about this.

Reply:
Ah you're right, thought it was there for transaction digest algo but in fact that's for second-stage htlc transactions fees, so yes right now we have to trust it.